src/datatype/signature.rs

```diff
@@ -130,51 +101,35 @@ impl SignatureType {
-        let pub_sig = "BusPdTkDUUG6ISM83snpKt0U8W3cKT8itiTXzJmWLHYBTytwCgW70gTY7z876cj6pELil9nBNC0YIdR4jhy4mu0cEiIIXqN3YWaEzMXAXt+QyjNBzm9POY6y3NF/jvpbZY3wgu5GbOkC6opRPoVARR3K79D9vRIEp6KDLbytqrFuDYd2rSRVfKkbdmt2kCJ17fV6NUWQocseil0kOtKZUI6jz/wt6M3pm5ni1I94Y+mjbwhw6LYEa9qQ++hwg875daaJSkb5jDJ0jkvwb28gsHLjZd1ldX0fuN3alV0RAb91m6klJbxu1ZC1QAfBsK2plkPKH+E4sOhu7NgneHqgFQ==";
+    fn test_rsa_sign_and_verify() {
+        let priv_key = Util::read_file("tests/keys/rsa.der").expect("read priv");
+        let pub_key = pem::parse(Util::read_file("tests/keys/rsa.pub").expect("read pub")).expect("pub_key").contents;
         let msg = b"hello";
-        let key = pem::parse(pub_key).unwrap().contents;
-        let sig = base64::decode(pub_sig).unwrap();
-        assert!(SignatureType::RsaSsaPss.verify_msg(msg, &key, &sig));
-
-        let mut bad_msg = msg.clone();
-        bad_msg[0] = msg[0] ^ 0x01;
-        assert_eq!(false, SignatureType::RsaSsaPss.verify_msg(&bad_msg, pub_key.as_bytes(), &sig));
-
-        let mut bad_sig = sig.clone();
-        bad_sig[0] = sig[0] ^ 0x01;
-        assert_eq!(false, SignatureType::RsaSsaPss.verify_msg(msg, pub_key.as_bytes(), &bad_sig));
+        let sig = SignatureType::RsaSsaPss.sign_msg(msg, &priv_key).expect("sign_msg");
+        let bad_msg = msg.iter().enumerate().map(|(i, b)| if i==0 { b^1 } else { *b }).collect::<Vec<_>>();
+        let bad_sig = sig.iter().enumerate().map(|(i, b)| if i==0 { b^1 } else { *b }).collect::<Vec<_>>();
+        assert!(SignatureType::RsaSsaPss.verify_msg(msg, &pub_key, &sig));
+        assert_eq!(false, SignatureType::RsaSsaPss.verify_msg(&bad_msg, &pub_key, &sig));
+        assert_eq!(false, SignatureType::RsaSsaPss.verify_msg(msg, &pub_key, &bad_sig));
     }
 
     #[test]
     fn test_ed25519_verify() {
-        // signing key: 0wm+qYNKH2v7VUMy0lEz0ZfOEtEbdbDNwklW5PPLs4WpCLVDpXuapnO3XZQ9i1wV3aiIxi1b5TxVeVeulbyUyw==
-        let pub_key = "qQi1Q6V7mqZzt12UPYtcFd2oiMYtW+U8VXlXrpW8lMs=";
-        let pub_sig = "/VniTdrxQlEXcx5QJGHqI7ptGwTq1wBThbfflb8SLRrEE4LQMkd5yBh/PWGvsU7cFNN+PNhFUZY4QwVq9p4MAg";
+        // priv_key: 0wm+qYNKH2v7VUMy0lEz0ZfOEtEbdbDNwklW5PPLs4WpCLVDpXuapnO3XZQ9i1wV3aiIxi1b5TxVeVeulbyUyw==
+        let key = base64::decode("qQi1Q6V7mqZzt12UPYtcFd2oiMYtW+U8VXlXrpW8lMs=").expect("parse key");
+        let sig = base64::decode("/VniTdrxQlEXcx5QJGHqI7ptGwTq1wBThbfflb8SLRrEE4LQMkd5yBh/PWGvsU7cFNN+PNhFUZY4QwVq9p4MAg").expect("parse sig");
         let msg = b"hello";
-        let key = base64::decode(pub_key).unwrap();
-        let sig = base64::decode(pub_sig).unwrap();
+        let bad_msg = msg.iter().enumerate().map(|(i, b)| if i==0 { b^1 } else { *b }).collect::<Vec<_>>();
```

Comment:
I think this is less clear than the other bit flipping thing I did previously. I would change this back for clarity's sake, but it's not enough for me to block the PR.